openedx/core/djangoapps/content_libraries/rest_api/libraries.py

```
@@ -790,6 +795,81 @@ def get(self, request, lib_key_str):
+
+    **POST Parameters**
+
+        A POST request must include the following parameters.
+
+        * file: (required) The backup file to restore the library from. Must be a
+          .zip file.
+
+    **GET Parameters**
+
+        A GET request must include the following parameters.
+
+        * task_id: (required) The UUID of a restore task.
+    """
+    def post(self, request):
+        """
+        Restore a library from a backup file.
+        """
+        if not request.user.has_perm(permissions.CAN_CREATE_CONTENT_LIBRARY):
+            raise PermissionDenied
+
+        if 'file' not in request.FILES:
+            raise ValidationError({'file': 'This field is required.'})
+
+        upload = request.FILES['file']
+        if not upload.name.endswith('.zip'):
+            raise ValidationError({'file': 'Archive must be a .zip file.'})
+
+        storage_path = course_import_export_storage.save(f'library_restore/{upload.name}', upload)
+
+        log.info("Learning package archive upload %s: Upload complete", upload.name)
+
+        restore_task = restore_library.delay(request.user.id, storage_path)
+
+        return Response({
+            'task_id': restore_task.task_id
+        })
+
+    def get(self, request):
+        """
+        Check the status of a library restore task.
+        """
+        task_id = request.GET.get('task_id')
+        if not task_id:
+            raise ValidationError({'task_id': 'This field is required.'})
+
+        task_status = get_object_or_404(UserTaskStatus, task_id=task_id)
+
+        artifact_name = ''
+
+        if task_status.state == UserTaskStatus.SUCCEEDED:
+            artifact_name = 'Library Restore'
+        elif task_status.state == UserTaskStatus.FAILED:
+            artifact_name = 'Error'
+
+        artifact = task_status.artifacts.filter(name=artifact_name).first()
+
+        return Response({
+            'state': task_status.state,
+            'result': json.loads(artifact.text) if artifact else {}
```

should we validate that artifact.text is a valid json? or just let it fail?

---

Yes, I also think we should validate if the artifact exists.